digital-workspace/latest/develop/extensionmechanism.md

```diff
@@ -0,0 +1,501 @@
+
+Once done, in the projects/my-extension path you will find the following structure:
+
+src folder containing all the typescript source code. Very important is the public-api.ts file defining all the inclusions of the extension and the lib/my-extension.module.ts file defining the module class for the extension.
+
+README.md file for documentation purposes as well as other files used for testing and configuration.
+
+To complete the creation, build the extension launching the following command.
+
+ng build my-extension
+Developing the basics of the ACA extension
+Now that the my-extension is created, let's add the proper configuration to the extension module. For this purpose, edit the projects/my-extension/src/lib/my-extension.module.ts file changing what is described below.
+
+// Add the import as described below.
+import { ExtensionService } from '@alfresco/adf-extensions';
+
+// Add the constructor as described below.
+NgModule({...})
+export class MyExtensionModule {
+  constructor(extensions: ExtensionService) {
+    extensions.setComponents({
+        'my-extension.main.component': MyExtensionComponent,
+    });
+  }
+}
+
+It's now time for the configuration of the brand new extension. For this purpose, you are going instruct the extension to add a link that you can see on the left menu of the landing page of ACA.
+
+To create the proper configuration, create the folder below in the described path.
+
+projects/my-extension/assets
+Once done, create the file projects/my-extension/assets/my-extension.json file with the following content.
+
+```json
+{
+  "$schema": "../../../extension.schema.json",
+  "$id": "my-extension",
+  "$version": "1.0.0",
+  "$vendor": "Your name or company name",
+  "$name": "plugin1",
+  "$description": "demo plugin",
+  "$license": "MIT",
+
+  "routes": [
+    {
+      "id": "my.extension.route",
+      "path": "ext/my/route",
+      "component": "my-extension.main.component"
+    }
+  ],
+
+  "features": {
+    "navbar": [
+      {
+        "id": "my.extension.nav",
+        "items": [
+          {
+            "id": "my.extension.main",
+            "icon": "extension",
+            "title": "My Extension",
+            "route": "my.extension.route"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+This is a very basic example, adding a “My Extension” item to the existing left menu, implementing a blank page containing “my-extension works!“ text appearing in the ACA landing page. From here, you can enrich the capabilities of your extension following the documentation at https://alfresco-content-app.netlify.app/#/extending/.
+
+Making the extension as part of the ACA application
+Now that the ACA extension is developed in its initial version, let's add the extension module to the list of the ones used by the application. To complete the task, edit the src/app/extensions.module.ts file as described below.
+
+// Add the following import to the page.
+
+```java
```

Comment:
```suggestion
```ts
```

Reply:
I'll confirm this change with Martin. I have made a note of it.